magefile.go

```diff
@@ -104,32 +106,66 @@ func Fields() error {
 }
 
 // -----------------------------------------------------------------------------
+
 // Customizations specific to apm-server.
 // - readme.md.tmpl used in packages is customized.
 // - apm-server.reference.yml is not included in packages.
+// - ingest .json files are included in packaging
+
+var ingestDirGenerated = filepath.Clean("build/packaging/ingest")
 
 func customizePackaging() {
 	var (
 		readmeTemplate = mage.PackageFile{
 			Mode:     0644,
 			Template: "package-README.md.tmpl",
 		}
+		ingestTarget = "ingest"
+		ingest       = mage.PackageFile{
+			Mode:   0644,
+			Source: ingestDirGenerated,
+		}
 	)
 	for idx := len(mage.Packages) - 1; idx >= 0; idx-- {
 		args := mage.Packages[idx]
-		switch args.Types[0] {
+		pkgType := args.Types[0]
+		switch pkgType {
+
 		case mage.Zip, mage.TarGz:
 			// Remove the reference config file from packages.
 			delete(args.Spec.Files, "{{.BeatName}}.reference.yml")
 
 			// Replace the README.md with an APM specific file.
 			args.Spec.ReplaceFile("README.md", readmeTemplate)
+			args.Spec.Files[ingestTarget] = ingest
+
 		case mage.Deb, mage.RPM:
 			delete(args.Spec.Files, "/etc/{{.BeatName}}/{{.BeatName}}.reference.yml")
 			args.Spec.ReplaceFile("/usr/share/{{.BeatName}}/README.md", readmeTemplate)
+			args.Spec.Files["/usr/share/{{.BeatName}}/"+ingestTarget] = ingest
 
 		case mage.DMG:
 			mage.Packages = append(mage.Packages[:idx], mage.Packages[idx+1:]...)
+
+		default:
+			panic(errors.Errorf("unhandled package type: %v", pkgType))
```

Comment:
why should packaging ever panic, a log is not enough?
if adding a `default`, would make sense to drop the  `case mage.DMG` then?

Reply:
The case should not be hit, but if additional packages are added to the `package.yml`, where we don't have a case defined, then we would be notified with this.